brickBreaker.js

```diff
@@ -0,0 +1,217 @@
+          brickLength,
+          colors[col]
+        );
+        this.bricks[row][col] = brick;
+      }
+    }
+  }
+
+  initializeEntities() {
+    this.score = 0;
+    this.ball = new Ball(this.ctx, this.canvas.width / 2, 500, 10, "#FFF", 4);
+    this.paddle = new Paddle(
+      this.canvas,
+      this.ctx,
+      100,
+      10,
+      150,
+      525,
+      5,
+      "grey"
+    );
+    this.initializeBricks();
+  }
+
+  initializeEventListeners() {
+    window.addEventListener("keydown", this.keyDownHandler.bind(this));
+    window.addEventListener("keyup", this.keyUpHandler.bind(this));
+  }
+
+  keyDownHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(-1);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(1);
+        break;
+      case " ":
+        this.reset();
+        break;
+    }
+  }
+
+  keyUpHandler(e) {
+    switch (e.key) {
+      case "ArrowLeft":
+        this.paddle.setDirection(0);
+        break;
+      case "ArrowRight":
+        this.paddle.setDirection(0);
+        break;
+    }
+  }
+
+  updateEntities() {
+    this.paddle.move();
+    this.ball.move();
+  }
+
+  detectCanvasCollision(ballTop, ballLeft, ballRight) {
+    if (ballLeft <= 0 || ballRight >= this.canvas.width) {
+      this.ball.invertXDirection();
+    } else if (ballTop <= 0) {
+      this.ball.invertYDirection();
+    } else if (ballTop > this.canvas.height) {
+      this.ball.xDirection = 0;
+      this.ball.yDirection = 0;
+    }
+  }
+
+  detectBallCollisionWithEntity(
+    entity,
+    ballTop,
+    ballBottom,
+    ballLeft,
+    ballRight
+  ) {
+    if (entity === this.paddle) {
+      const paddleTop = this.paddle.getTopEdge();
+      const paddleBottom = this.paddle.getBottomEdge();
+      const paddleLeft = this.paddle.getLeftEdge();
+      const paddleRight = this.paddle.getRightEdge();
+
+      if (paddleTop < ballBottom && ballTop >= paddleBottom) {
+        if (paddleLeft <= ballLeft && paddleRight >= ballRight) {
+          this.ball.invertYDirection();
+        }
+      }
+    } else if (entity === this.bricks) {
+      for (let row = 0; row < this.brickRows; row++) {
+        for (let col = 0; col < this.brickCols; col++) {
+          const brick = this.bricks[row][col];
+          const brickTop = brick.getTopEdge();
+          const brickBottom = brick.getBottomEdge();
+          const brickLeft = brick.getLeftEdge();
+          const brickRight = brick.getRightEdge();
+          if (!brick.broken) {
+            if (ballRight >= brickLeft && ballLeft <= brickRight) {
+              if (brickBottom >= ballTop || brickTop >= ballBottom) {
+                this.ball.invertXDirection();
+                this.ball.invertYDirection();
+                brick.broken = true;
+                this.score += 1;
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  detectCollisions() {
+    const ballTop = this.ball.getTopEdge();
```

Comment:
nit: not sure what the point of these are when you could just call this.ball.getTopEdge() in the collision logics instead